src/julia_gc.c

```diff
@@ -527,6 +528,9 @@ void GapRootScanner(int full)
     // current_task != root_task.
     char * stackend = (char *)jl_task_stack_buffer(task, &size, &tid);
     stackend += size;
+    if (JuliaTLS->tid == 0 && JuliaTLS->root_task == task) {
+        stackend = (char *)GapStackBottom;
```

Comment:
Hmm, but note that this only will work right if GAP is the main process, loading Julia later. If, however, one is starting Julia and from there loading GAP, then `GapStackBottom ` may not be at the base of the stack frame.

This is closely related to (part of) the discussion on issue #3089 ;  see also PR #3096

Reply:
Yes, and `GapTaskScanner()` may also need an update. That said, it should properly be handled so that Julia gets to know the proper end of the stack for task switching.

Reply:
ISTM there needs to be a `SetStackBottomBags` (as in #3096) for Julia GC as well.

Reply:
Possible, but this code is very much in flux, so it doesn't really matter at this stage.